site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7615,4 +7615,26 @@ private function getGradeableMinutesOverride(string $gradeable_id): array {
         $this->course_db->query('SELECT * FROM gradeable_allowed_minutes_override WHERE g_id=?', [$gradeable_id]);
         return $this->course_db->rows();
     }
+
+    /**
+     * Gets the number of students who have submitted to a given gradeable
+     *
+     * @param string $gradeable_id
+     * @return int
+     */
+    public function getTotalStudentsWithSubmissions(string $gradeable_id): int {
+        $this->course_db->query('SELECT DISTINCT user_id FROM electronic_gradeable_data WHERE g_id=?', [$gradeable_id]);
+        return count($this->course_db->rows());
+    }
+
+    /**
+     * Gets the total number of submissions made to a given gradeable
+     *
+     * @param string $gradeable_id
+     * @return int
+     */
+    public function getTotalSubmissionsToGradeable(string $gradeable_id): int {
+        $this->course_db->query('SELECT * FROM electronic_gradeable_data WHERE g_id=?', [$gradeable_id]);
+        return count($this->course_db->rows());
+    }
```

Comment:
You should be doing the `count` within the query itself, not within PHP. This will save on both memory and speed.